File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    int N = s.size();
    
    for (int i = 0; i < N; i++) {
        if (s[i] >= 65 && s[i] <= 90) {
            answer += (s[i] - 'A' + n) % 26 + 'A';
        } else if (s[i] >= 97 && s[i] <= 122) {
            answer += (s[i] - 'a' + n) % 26 + 'a';
        } else {
            answer += s[i];
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = []
    
    l = list(s)
    for c in l:
        C = ord(c)
        if (C >= 65 and C <= 90):
            answer.append(chr((C - 65 + n) % 26 + 65))
        elif (C >= 97 and C <= 122):
            answer.append(chr((C - 97 + n) % 26 + 97))
        else:
            answer.append(c)
    
    return "".join(answer)